src/main.py

```diff
@@ -109,59 +109,68 @@ def run(options: Options) -> int:
         struct_field_inference=options.struct_field_inference,
     )
     global_info = GlobalInfo(asm_data, function_names, typemap, typepool)
-    function_infos: List[Union[FunctionInfo, Exception]] = []
-    for function in functions:
-        try:
-            info = translate_to_ast(function, options, global_info)
-            function_infos.append(info)
-        except Exception as e:
-            # Store the exception for later, to preserve the order in the output
-            function_infos.append(e)
-
-    return_code = 0
-    try:
-        if options.visualize_flowgraph:
-            fn_info = function_infos[0]
-            if isinstance(fn_info, Exception):
-                raise fn_info
-            print(visualize_flowgraph(fn_info.flow_graph))
-            return 0
-
-        if options.structs:
-            type_decls = typepool.format_type_declarations(fmt)
-            if type_decls:
-                print(type_decls)
-
-        global_decls = global_info.global_decls(fmt, options.global_decls)
-        if global_decls:
-            print(global_decls)
-    except Exception as e:
-        print_exception_as_comment(
-            e, context=None, sanitize=options.sanitize_tracebacks
-        )
-        return_code = 1
 
-    for index, (function, function_info) in enumerate(zip(functions, function_infos)):
-        if index != 0:
-            print()
-        try:
-            if options.print_assembly:
-                print(function)
-                print()
+    for pass_number in range(options.passes):
```

Comment:
(if options.passes <= 0 we'll crash by referring to return_code while it's uninitialized. I bet someone's going to do it. :p)

not sure if the "is_final_pass" thing is actually cleaner than the loop up to options.passes-1 that you had before.

Reply:
I'll revert to the loop option.